### PR TITLE
[Messages] Reception and confirmation times

### PR DESCRIPTION
**Pull request by odesenfans**
Added two new fields to the messages collection. We now store
the initial time of reception of a message in the `reception_time`
field, and the earliest on-chain confirmation time in
the `confirmation_time` field.

These fields will be used to order messages more precisely/securely
and to compute metrics, for example the propagation time of a message
on the network.

The `time` field of messages is now deprecated, as it is defined
by users and not signed. If you need the user time, use `content.time`.

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #265.

File: tests/chains/test_confirmation.py [hunk 1]
```diff
@@ -102,13 +109,17 @@ async def test_process_confirmed_message(test_db):
...
+        message, reception_time=reception_time, tx_context=tx_context
+    )
 
     # Now, confirm the message
     message_in_db = await Message.collection.find_one({"item_hash": item_hash})
 
     assert message_in_db is not None
     assert message_in_db["confirmed"]
+    assert message_in_db["confirmation_time"] == tx_context.time
+    assert message_in_db["reception_time"] == reception_time
```
[the remark this answers is not in the document]
Comment (odesenfans) [reply tests/chains/test_confirmation.py:122]: Yes, I'll add them.

File: src/aleph/schemas/validated_message.py [hunk 2]
```diff
@@ -159,10 +175,18 @@ def make_message_upsert_query(message: BaseValidatedMessage[Any, Any]) -> Dict:
...
     }
 
     # Add fields related to confirmations
-    updates.update(make_confirmation_update_query(message.confirmations))
+    confirmation_updates = make_confirmation_update_query(message.confirmations)
+    for k, v in confirmation_updates.items():
+        try:
+            updates[k].update(v)
+        except KeyError:
+            updates[k] = v
```
[the remark this answers is not in the document]
Comment (hoh) [reply src/aleph/schemas/validated_message.py:190]: Oh, I see. Yes, there should be a cleaner way. Else at least a dedicated testable function to do just that.

File: tests/chains/test_confirmation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hoh) [reply tests/chains/test_confirmation.py:122]: https://github.com/aleph-im/aleph-message/issues/31

**Pull request closed without merging**